pkg/http-client/client.go

```diff
@@ -0,0 +1,194 @@
+	GraphQLRetrieveSubjectResults(ctx context.Context, gitoid string) (api.RetrieveSubjectResults, error)
+	GraphQLRetrieveSearchResults(ctx context.Context, algo string, digest string) (api.SearchResults, error)
+	GraphQLQueryIface(ctx context.Context, query string, variables interface{}) (*GraphQLResponseInterface, error)
+	GraphQLQueryToDst(ctx context.Context, query string, variables interface{}, dst interface{}) error
+	GraphQLQueryReadCloser(ctx context.Context, query string, variables interface{}) (io.ReadCloser, error)
+}
+
+func CreateArchivistaClient(httpClient *http.Client, baseURL string) (*ArchivistaClient, error) {
+	client := ArchivistaClient{
+		BaseURL: baseURL,
+		Client:  http.DefaultClient,
+	}
+	if httpClient != nil {
+		client.Client = httpClient
+	}
+	var err error
+	client.GraphQLURL, err = url.JoinPath(client.BaseURL, "query")
+	if err != nil {
+		return nil, err
+	}
+	return &client, nil
+}
+
+func (ac *ArchivistaClient) DownloadDSSE(ctx context.Context, gitoid string) (dsse.Envelope, error) {
+	reader, err := api.DownloadReadCloserWithHTTPClient(ctx, ac.Client, ac.BaseURL, gitoid)
+	if err != nil {
+		return dsse.Envelope{}, err
+	}
+	env := dsse.Envelope{}
+	if err := json.NewDecoder(reader).Decode(&env); err != nil {
+		return dsse.Envelope{}, err
+	}
+	return env, nil
+}
+
+func (ac *ArchivistaClient) DownloadReadCloser(ctx context.Context, gitoid string) (io.ReadCloser, error) {
+	return api.DownloadReadCloserWithHTTPClient(ctx, ac.Client, ac.BaseURL, gitoid)
+}
+
+func (ac *ArchivistaClient) DownloadWithWriter(ctx context.Context, gitoid string, dst io.Writer) error {
+	return api.DownloadWithWriterWithHTTPClient(ctx, ac.Client, ac.BaseURL, gitoid, dst)
+}
+
+func (ac *ArchivistaClient) Store(ctx context.Context, envelope dsse.Envelope) (api.UploadResponse, error) {
+	return api.Store(ctx, ac.BaseURL, envelope)
+}
+
+func (ac *ArchivistaClient) StoreWithReader(ctx context.Context, r io.Reader) (api.UploadResponse, error) {
+	return api.StoreWithReader(ctx, ac.BaseURL, r)
+}
+
+type GraphQLRequestBodyInterface struct {
+	Query     string      `json:"query"`
+	Variables interface{} `json:"variables,omitempty"`
+}
+
+type GraphQLResponseInterface struct {
+	Data   interface{}
+	Errors []api.GraphQLError `json:"errors,omitempty"`
+}
+
+func (ac *ArchivistaClient) GraphQLRetrieveSubjectResults(
+	ctx context.Context,
+	gitoid string,
+) (api.RetrieveSubjectResults, error) {
+	return api.GraphQlQuery[api.RetrieveSubjectResults](
+		ctx,
+		ac.BaseURL,
+		api.RetrieveSubjectsQuery,
+		api.RetrieveSubjectVars{Gitoid: gitoid},
+	)
+}
+
+func (ac *ArchivistaClient) GraphQLRetrieveSearchResults(
+	ctx context.Context,
+	algo string,
+	digest string,
+) (api.SearchResults, error) {
+	return api.GraphQlQuery[api.SearchResults](
+		ctx,
+		ac.BaseURL,
+		api.SearchQuery,
+		api.SearchVars{Algorithm: algo, Digest: digest},
+	)
+}
+
+func (ac *ArchivistaClient) GraphQLQueryIface(
```

Comment:
Is this function intended to be used by a user? It would probably be helpful to have some descriptions above the functions, that way they will show up in https://pkg.go.dev/